platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+// if this is genesis, or if this is a TSS upgrade of an existing network.
+    }
+
+	/*
+	// This block will be uncommented once the TSS protocol is implemented
+	if (Candidate Roster is not complete) {
+		// Need to populate the Candidate Roster with TSS key material and signatures.
+		// Call the "TSS State Observer" to make it initiate the TSS protocol.
+		// See the Detecting a new Candidate Roster (aka the new “Platform API”) section below.
+		callTSSStateObserverToStartTSSProtocol();
+		// NOTE: processing of TSS messages will accumulate TSS key material
+		// in Candidate Roster and will emit a percentage metric indicating the readiness
+		// of the Candidate Roster for adoption. Once the metric is at 100%, this shows
+		// that the network can be restarted in order to adopt the Candidate Roster.
+	}
+	*/
+            }
+
+            if(
+the State
+has no
+Active Roster){
+
+// This should never happen
+throwFatalErrorAndShutdown();
+}
+
+// Check if the Active Roster is TSS-enabled
+        if(
+Active Roster
+is not
+TSS-enabled){
+  /*
+  // This block will be uncommented once the TSS protocol is implemented
+	if (isGenesis) {
+		rejectAnythingButTSSMessages = true;
+		// Note that a TSS upgrade of an existing network
+		// shouldn't disable non-TSS messages processing.	
+	}
+	*/
+
+        // In Genesis, we've just disabled processing anything but TSS. So a new
+        // network will eventually become TSS-enabled and only then will start
+        // processing non-TSS events.
+        // An existing, non-genesis network will continue to operate as before,
+        // and if a Candidate Roster was present in the state, we enabled processing TSS messages
+        // above. So it will become TSS-enabled eventually, too, indicating
+        // the readiness for a TSS upgrade via the metric mentioned above.
+        // Note that until a Candidate Roster is installed in the state, the existing network
+        // will simply continue to operate as usual with its regular RSA keys.
+        }
+
+// At this point the Active Roster in the state is what we'll be using as a roster.
+// If an incomplete Candidate Roster exists, we've started exchanging TSS messages above.
+// If it's genesis, we've disabled processing anything but TSS messages above.
+// If Active Roster is already TSS-enabled (a mature network running after the TSS
+// upgrade), then the node will be able to sign using TSS.
+// If Active Roster isn't TSS-enabled yet, then the node will sign using its RSA key only,
+// just as it does today.
+```
+
+### Bootstrapping Genesis for a brand new network
+
+This proposal already creates a new interface for the services to install a Candidate Roster—via a Platform API, or
+similar—which ultimately stores the Candidate Roster in a designated location in the State.
+The Startup procedure described above consumes the Candidate Roster from that location and starts its processing or
```

Comment:
There's a complication here that the above pseudo code is not factoring in.  

If we don't do inversion of control to where Services parses `config.txt` then we have to explain how services gets the fields they need from config.txt.   Rosters do not have `memo` field which currently contains the Hedera account number for the node.  

It's easier to just do inversion of control and have the App set a genesis roster.  That way Services can get whatever they need from config.txt